### PR TITLE
chore: Update default vale version `3.4.1`

### PR DESCRIPTION
**Pull request by dipinknair**
Update the default vale version to `3.4.1`

### CODE REVIEW

**Comment by dipinknair** [comment]
@ansys-reviewer-bot review

**Comment by jorgepiloto** [comment]
Before merging this, can we check for #411.

**Comment by RobPasMue** [comment]
> Before merging this, can we check for #411?

Can't we merge both PRs into one?

**Comment by RobPasMue** [comment]
I believe we should leave this PR unmerged until we decide to actually do a v6

**Comment by jorgepiloto** [comment]
I see... That's a good point. Otherwise we may run into some v6 -> v7

**Comment by RobPasMue** [comment]
@dipinknair - could you adapt the docs/source/migration_guide.rst file to reflect the changes needed to be done?

**Comment by Revathyvenugopal162** [comment]
vale has upgraded to version "3.3.0," and everything seems fine. I think we can go straight to version 3.3 instead of considering version 3.1.

File: doc-style/action.yml [hunk 1]
```diff
@@ -29,9 +29,7 @@ inputs:
   vale-version:
     description: >
       Version number for Vale.
-    # TODO: use the latest stable version from Vale, see issue
-    # https://github.com/ansys/actions/issues/350
-    default: '2.29.6'
+    default: '3.1.0'
```
Comment (RobPasMue) [inline doc-style/action.yml:32]:
```suggestion
    default: '3.3.0'
```

File: doc/source/migration_guide.rst [hunk 2]
```diff
@@ -12,7 +12,8 @@ Development version
 
 **Breaking Changes:**
 
-- N/A
+- Upgrade default ``vale`` version from ``2.29.6`` to ``3.1.0`` in ``ansys/actions/doc-style`` action.
```
Comment (RobPasMue) [inline doc/source/migration_guide.rst:15]:
```suggestion
- Upgrade default ``vale`` version from ``2.29.6`` to ``3.3.0`` in ``ansys/actions/doc-style`` action.
```

File: doc-style/action.yml [hunk 3]
```diff
@@ -51,9 +51,7 @@ inputs:
   vale-version:
     description: >
       Version number for Vale.
-    # TODO: use the latest stable version from Vale, see issue
-    # https://github.com/ansys/actions/issues/350
-    default: '2.29.6'
+    default: '3.3.0'
```
Comment (jorgepiloto) [inline doc-style/action.yml:54]: @dipinknair, should we update this to `3.4.1`? This is so far the latest version in Vale. If so, feel free to update and merge this. We are likely to release v6 next week.

**Review by jorgepiloto — APPROVED** [review]

File: doc/source/migrations/index.rst [hunk 4]
```diff
@@ -22,7 +22,8 @@ Development version
 
 **Breaking changes:**
 
-- N/A
+- Upgrade default ``vale`` version from ``2.29.6`` to ``3.3.0`` in ``ansys/actions/doc-style`` action.
+  Vale config file ``.vale.ini`` and ``Vocab/ANSYS`` has to be changed as per `Vale section <https://dev.docs.pyansys.com/doc-style/formatting-tools.html#vale>`_ in PyAnsys developer’s guide
 
 **Migration steps:**
```
Comment (RobPasMue) [inline doc/source/migrations/index.rst:28]: @dipinknair - add a migration steps guide for moving to version 3 of vale.

**Review by RobPasMue — CHANGES_REQUESTED** [review]

File: doc/source/migrations/index.rst (hunk 4 above)
Comment (dipinknair) [reply doc/source/migrations/index.rst:28]: @RobPasMue  Added ! Please review.

File: doc/source/migrations/docs-style-vale-version-update.rst [hunk 5]
```diff
@@ -0,0 +1,48 @@
+.. _docs_style_vale_update:
+
+Doc-style action - vale update
+==============================
+
+When migrating ``ansys/actions/doc-style`` from v5 to higher versions, vale version is upgraded to ``v3``.
+Certain changes must be made to the repository to ensure that the ``doc-style`` action performs without issues.
+Additionally, it is possible to use ansys actions older than version 5 while utilizing vale version 3
```
Comment (RobPasMue) [inline doc/source/migrations/docs-style-vale-version-update.rst:8]:
```suggestion
Additionally, it is possible to use ``ansys/actions`` older than ``v5`` while utilizing Vale ``v3``
```

File: doc/source/migrations/docs-style-vale-version-update.rst [hunk 6]
```diff
@@ -0,0 +1,48 @@
+.. _docs_style_vale_update:
+
+Doc-style action - vale update
+==============================
+
+When migrating ``ansys/actions/doc-style`` from v5 to higher versions, vale version is upgraded to ``v3``.
```
Comment (RobPasMue) [inline doc/source/migrations/docs-style-vale-version-update.rst:6]:
```suggestion
When migrating ``ansys/actions/doc-style`` from ``v5`` to higher versions, the default Vale version is upgraded to ``v3``.
```

File: doc/source/migrations/docs-style-vale-version-update.rst [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+            uses: ansys/actions/doc-style@v5
+            with:
+            token: ${{ secrets.GITHUB_TOKEN }}
+            vale-version: "3.4.1"
+
+In any of the above two conditions there  are two changes needs to implemented:
+
+1. Update **vocabularies** path
+
+   In order to comply with the requirements of Vale version 3 or above, it is necessary to update the vocabularies
```
Comment (RobPasMue) [inline doc/source/migrations/docs-style-vale-version-update.rst:27]:
```suggestion
   In order to comply with the requirements of Vale ``v3``, it is necessary to update the vocabularies
```

File: doc/source/migrations/docs-style-vale-version-update.rst [hunk 8]
```diff
@@ -0,0 +1,48 @@
+.. _docs_style_vale_update:
+
+Doc-style action - vale update
+==============================
+
+When migrating ``ansys/actions/doc-style`` from v5 to higher versions, vale version is upgraded to ``v3``.
+Certain changes must be made to the repository to ensure that the ``doc-style`` action performs without issues.
+Additionally, it is possible to use ansys actions older than version 5 while utilizing vale version 3
+or above as input for the ``doc-style`` action like below.
```
Comment (RobPasMue) [inline doc/source/migrations/docs-style-vale-version-update.rst:9]:
```suggestion
as input for the ``doc-style`` action like below.
```

File: doc/source/migrations/docs-style-vale-version-update.rst [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+1. Update **vocabularies** path
+
+   In order to comply with the requirements of Vale version 3 or above, it is necessary to update the vocabularies
+   path under the ``doc/styles`` directory. Specifically, the default vocabularies path should be modified from
+   ``Vocab/ANSYS`` to ``config/vocabularies/ANSYS``. This adjustment ensures that Vale can locate the required vocabulary files.
+
+   .. note:: Update ``.gitignore``
+
+    If your repository has ``.gitignore`` files under ``styles`` folder, please update them according to vocabularies changes
+
```
Comment (RobPasMue) [inline doc/source/migrations/docs-style-vale-version-update.rst:34]:
```suggestion
   .. note:: Update ``.gitignore``

      If your repository has ``.gitignore`` files under ``styles`` folder, please update them according to vocabularies changes

```

File: doc/source/migrations/docs-style-vale-version-update.rst [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+2. Turn off **Vale Terms**
+
+   Locate the section in your ``doc/.vale.ini`` file where styles are applied.
+   Add ``Vale.Terms = NO`` under the section where styles are applied, typically marked with ``[*.{rst}]``.
+
+   .. code:: ini
+
+    [*.{rst}]
+    BasedOnStyles = Vale, Google
+    Vale.Terms = NO
```
Comment (RobPasMue) [inline doc/source/migrations/docs-style-vale-version-update.rst:44]:
```suggestion
   .. code:: ini

      [*.{rst}]
      BasedOnStyles = Vale, Google
      Vale.Terms = NO
```

File: doc/source/migrations/index.rst [hunk 11]
```diff
@@ -22,10 +22,12 @@ Development version
 
 **Breaking changes:**
 
-- N/A
+- Upgrade default ``vale`` version from ``2.29.6`` to ``3.4.1`` in ``ansys/actions/doc-style`` action.
+- Vale config file ``.vale.ini`` and ``Vocab/ANSYS`` has to be changed.
```
Comment (RobPasMue) [inline doc/source/migrations/index.rst:26]:
```suggestion
- Vale configuration file ``.vale.ini`` and ``Vocab/ANSYS`` has to be changed.
```

File: doc/source/migrations/index.rst [hunk 12]
```diff
@@ -79,8 +81,9 @@ Version ``v4``
   for a detailed migration guide.
 
 .. toctree::
-   :hidden:
-   :maxdepth: 3
+  :hidden:
+  :maxdepth: 3
```
Comment (RobPasMue) [inline doc/source/migrations/index.rst:85]:
```suggestion
   :hidden:
   :maxdepth: 3
```

File: doc/source/migrations/index.rst [hunk 13]
```diff
@@ -79,8 +81,9 @@ Version ``v4``
...
 
-   docs-changelog-setup
-   docs-deploy-changelog-setup
+  docs-style-vale-version-update
+  docs-changelog-setup
+  docs-deploy-changelog-setup
```
Comment (RobPasMue) [inline doc/source/migrations/index.rst:89]:
```suggestion
   docs-changelog-setup
   docs-deploy-changelog-setup
   docs-style-vale-version-update
```

File: doc/source/migrations/index.rst [hunk 14]
```diff
@@ -22,10 +22,12 @@ Development version
...
 
-- N/A
+- Upgrade default ``vale`` version from ``2.29.6`` to ``3.4.1`` in ``ansys/actions/doc-style`` action.
+- Vale config file ``.vale.ini`` and ``Vocab/ANSYS`` has to be changed.
 
 **Migration steps:**
 
+- To set up your repository to use the ``ansys/actions/doc-style`` action, see the :ref:`docs_style_vale_update`.
 - To set up your repository to use the ``ansys/actions/doc-changelog`` action, see the :ref:`docs_changelog_action_setup`.
 - To set up your repository to use the ``ansys/actions/doc-deploy-changelog`` action, see the :ref:`docs_deploy_changelog_action_setup`.
```
Comment (RobPasMue) [inline doc/source/migrations/index.rst:33]:
```suggestion
- To set up your repository to use the ``ansys/actions/doc-changelog`` action, see the :ref:`docs_changelog_action_setup`.
- To set up your repository to use the ``ansys/actions/doc-deploy-changelog`` action, see the :ref:`docs_deploy_changelog_action_setup`.
- To set up your repository to use the ``ansys/actions/doc-style`` action, see the :ref:`docs_style_vale_update`.

```

**Review by RobPasMue — COMMENTED** [review]
Some comments

**Review by RobPasMue — APPROVED** [review]

**Comment by RobPasMue** [comment]
All good @dipinknair - feel free to merge

**Pull request merged**